builtin/rebase.c

```diff
@@ -1408,6 +1413,12 @@ int cmd_rebase(int argc, const char **argv, const char *prefix)
 		}
 	}
 
+	if (options.ignore_whitespace && !options.git_am_opts.argc) {
+		options.strategy = options.strategy ?
+					options.strategy : "recursive";
+		imply_interactive(&options, "--interactive");
```

Comment:
We are not adding `REBASE_INTERACTIVE_EXPLICIT` so this won't open the editor which should be the default behaviour?

Reply:
That sounds correct.  I do however not quite understand why we need the `imply_interactive` call here.  All that seems to do is check that we don't use flags meant only for interactive rebase for another type of rebase (and set `opts->type` to `REBASE_INTERACTIVE` if necessary, which doesn't seem to be the case here?

So I think this whole hunk can be dropped, or am I missing something here?

Reply:
Oh! I remembered I added this to imply interactive because I was adding `--strategy` here itself, instead of passing it to `builtin/rebase--interactive.c` before. (And --strategy and am opts are incompatible). Surely I'll move the code and change the [documentation](https://github.com/r1walz/git/blob/f7a8057c5e2b0706bf5023005c7361e4fecbd589/Documentation/git-rebase.txt#L372-L373) too.